### PR TITLE
SOLR-16799 upgrade Solr to use Lucene 9.6.0

### PR DESCRIPTION
**Pull request by stillalex**
https://issues.apache.org/jira/browse/SOLR-16799

# Description

Lucene 9.6.0 upgrade. All tests seem to be passing.

* DocValuesTermsQuery is gone. this triggered some changes that will need a close look in TermsQParserPlugin and GraphEdgeCollector.

* TopLevelDocValuesTermsQuery  uses getTermData() which is deprecated, likely this part will need another rewrite

* Field storedValue() method was introduced as an optimization to detect value type:

    - this broke the EnumFieldType - org.apache.solr.TestDistributedSearch.test was failing. added details about the fix on the Jira ticket - https://issues.apache.org/jira/browse/SOLR-16799?focusedCommentId=17726446&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17726446
    - added handling for this to the LargeLazyField, needs a close look

* updated renovate config to skip lucene updates. not sure how to test this.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java [hunk 1]
```diff
@@ -539,6 +541,21 @@ public Reader readerValue() {
...
+      } else {
+        return new StoredValue(stringValue());
+      }
+    }
+
+    @Override
+    public InvertableType invertableType() {
+      // TODO SOLR-16799 when to use InvertableType.TOKEN_STREAM and when to use null?
+      return null;
+    }
```
Comment (tflobbe) [inline solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java:558]: It shouldn't matter, right? this is only for stored field retrieval, not for indexing. This should never be called AFAIK

File: solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java [hunk 2]
```diff
@@ -539,6 +541,21 @@ public Reader readerValue() {
...
       return null;
     }
+
+    @Override
+    public StoredValue storedValue() {
+      if (fieldType().stored() == false) {
+        return null;
+      } else {
+        return new StoredValue(stringValue());
+      }
```
Comment (tflobbe) [inline solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java:551]: I guess `fieldType().stored()` will never be `false` here?

**Review by tflobbe — COMMENTED** [review]
LGTM. A couple minor comments added

File: solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java (hunk 2 above)
Comment (stillalex) [reply solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java:551]: I think you are correct. will remove the if block

File: solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java (hunk 1 above)
Comment (stillalex) [reply solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java:558]: hmm. I agree, will remove the TODO

**Comment by stillalex** [comment]
thank you for talking a look @tflobbe! I updated the PR.

**Review by tflobbe — APPROVED** [review]
LGTM

**Comment by tflobbe** [comment]
@stillalex, would you add a CHANGES entry for 9.3?

**Comment by stillalex** [comment]
sorry for the delay @tflobbe! pushed a changes.txt update. I rebased too, running the tests again on my local just for good measure.

**Pull request merged**